### ISSUE TITLE
Settings search with "bad" internet is blocked

### ISSUE THREAD

**Issue by alexandrudima** [Microsoft/vscode#42741]
Simulate "bad" internet:

![image](https://user-images.githubusercontent.com/5047891/35684994-bcecd342-0768-11e8-9ce4-0dc7e6e3ed31.png)

Search for settings in the Settings Editor. It will search for parts of your query, then block for 20s, then search for the rest of the query.

fyi @egamma

**Comment by sandy081** [comment]
@roblourens I provided a fix here to separate local search and remote search. So when remote search is blocked or taking too much of time, local search can still run and show results. 

But I see few issues with remote search

1. Remote search is not cancelled when I cleared the search input
2. Showing outdated results. See in the following video, remote search is showing outdated results when filter string is cleared.
3. Can we have a timeout for remote search? For eg., cancel the requests if they take more than one second?

![kapture 2018-02-01 at 22 10 06](https://user-images.githubusercontent.com/10746682/35703648-2109d8d2-079d-11e8-9ec2-94f2e24f8b4a.gif)

**Comment by roblourens** [comment]
We do have a timeout of 5 seconds, but @alexandrudima bypassed it because it's implemented by XHR. So I think that isn't a good test.

**Comment by roblourens** [comment]
But I'll look at @sandy081's change

**Comment by roblourens** [comment]
I see that the earlier progress bar change caused the local search to wait on the remote search, and the new change decouples them, which is correct. 

I debugged some and realized that I'm doing something slightly wrong, in that the remote search is debounced, but each new search waits on the old one to complete. It should actually fire immediately and cancel the previous search if it has not completed yet. I didn't realize that the ThrottledDelayer is queuing the promises, makes sense though.

I have a change to make the ThrottledDelayer simply not wait on the search to complete. But, I don't think I want to check it in now because the impact isn't that bad, with the remote search timeout. And it isn't noticeable under normal network conditions. The change is here: https://github.com/Microsoft/vscode/commit/0dbb7a0ee4f3c2aa4e469f07787e7e37d1f8f65c

**Comment by sandy081** [comment]
@roblourens Good. 

Opened a separate issue to improve this throttled behaviour. 

Closing this one as decoupling local and remote searches fixes this issue.

**Comment by alexandrudima** [comment]
Thank you!
